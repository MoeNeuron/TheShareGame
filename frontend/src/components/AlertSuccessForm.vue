<template>
   <b-row class="justify-content-md-center">
    <b-alert show variant="success"
             v-if="show">{{ msg }}
    </b-alert>
  </b-row>
</template>

<script>

import { BRow, BAlert } from 'bootstrap-vue';

export default {
  components: { BRow, BAlert },
  name: 'AlertSuccessForm',
  props: ['show', 'msg'],
};
</script>

<style scoped>

</style>
