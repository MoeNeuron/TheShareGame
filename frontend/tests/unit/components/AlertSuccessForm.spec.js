import { createLocalVue, mount } from '@vue/test-utils';

import AlertSuccessForm from '@/components/AlertSuccessForm.vue';

const localVue = createLocalVue();

describe('AlertSuccessForm', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(AlertSuccessForm, {
      localVue,
      propsData: {
        show: true,
        msg: 'Form Submitted!',
      },
    });
  });

  it('only shows when show is true', () => {
    wrapper = mount(AlertSuccessForm, {
      localVue,
      propsData: {
        show: false,
        msg: 'Form Submitted!',
      },
    });

    const html = wrapper.html();

    expect(html).not.toContain('Form Submitted!');
  });

  it('matches Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
