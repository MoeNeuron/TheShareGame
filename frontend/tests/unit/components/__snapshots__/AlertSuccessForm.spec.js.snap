// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertSuccessForm matches Snapshot 1`] = `
<div class="row justify-content-md-center">
  <div role="alert" aria-live="polite" aria-atomic="true" class="alert alert-success">
    <!---->Form Submitted!
  </div>
</div>
`;
